### PR TITLE
[MRG] Implement map_surface parameter for Brain.add_foci

### PR DESCRIPTION
**Pull request by wmvanvliet**
When adding foci to a plotted brain, the `map_surface` parameter allows the user to project them onto a mesh. This was not implemented in the latest master yet, so I copied over [the relevant code from PySurfer](https://github.com/nipy/PySurfer/blob/master/surfer/viz.py#L1688-L1697). Seems to work nicely. Needs a test maybe?

### CODE REVIEW

**Comment by wmvanvliet** [comment]
Before, `map_surface=None`:

fig1


After, `map_surface='white'`:

fig2

**Comment by larsoner** [comment]
I think map_surface=None should use the current surface (which here would be white) and this produce the same result, unless I'm missing something. Also maybe we don't even need the parameter and should always use the currently loaded surface?

**Comment by wmvanvliet** [comment]
I don't think you'd always want to map to a surface. At this point, the behavior is the same as that of PySurfer.

**Comment by larsoner** [comment]
Looks reasonable, just needs a test and `latest.inc` entry

**Comment by agramfort** [comment]
Works for me

**Comment by wmvanvliet** [comment]
Hrm, how *do* I test this?

**Comment by larsoner** [comment]
At the very least a smoke test that you can `add_foci(..., map_surface=...)` does not error is a good start

Another low-effort but more complete option would be to `logger.debug` the obtained coordinates, `with mne.utils.catch_logging() as log:` during your call, and ensure that something sensible shows up `in log.getvalue()`

**Comment by wmvanvliet** [comment]
I've made it so the coordinates of the foci are tracked in `Brain._data[hemi]['foci']`. It may actually be useful for users to be able to retrieve the coordinates of the projected foci. Test added.

**Comment by wmvanvliet** [comment]
Apparently, I'm doing something wrong with this test. Many platforms report they cannot load anything 3D viz related. But how come the other tests in `test_brain.py` work?

File: mne/viz/_brain/tests/test_brain.py [hunk 1]
```diff
@@ -1010,3 +1011,12 @@ def _create_testing_brain(hemi, surf='inflated', src='surface',
         clim=clim, src=sample_src,
         **kwargs)
     return brain_data
+
+
+def test_foci_mapping(tmp_path):
```
Comment (larsoner) [inline mne/viz/_brain/tests/test_brain.py:1016]:
```suggestion
def test_foci_mapping(tmp_path, renderer_interactive_pyvistaqt):
```

**Comment by larsoner** [comment]
Failures look real, does `pytest mne/viz/_brain -k test_brain_traces` pass for you locally?

**Comment by larsoner** [comment]
Failures look real

https://github.com/mne-tools/mne-python/runs/5180078196?check_suite_focus=true#step:13:3433

Do tests pass locally?

**Review by agramfort — APPROVED** [review]

**Comment by wmvanvliet** [comment]
Programming is hard. Ok, tests all pass. This is ready for merge!

**Comment by larsoner** [comment]
Thanks @wmvanvliet !

**Comment by larsoner** [comment]
This appears to have broken CircleCI:
```
WARNING: /home/circleci/project/tutorials/forward/50_background_freesurfer_mne.py failed to execute correctly: Traceback (most recent call last):
  File "/home/circleci/project/tutorials/forward/50_background_freesurfer_mne.py", line 455, in <module>
    brain.add_foci(xyz, hemi='lh', color='k')
  File "/home/circleci/project/mne/viz/_brain/_brain.py", line 2621, in add_foci
    if 'foci' in self._data[hemi]:
KeyError: 'lh'
```
@wmvanvliet I can replicate locally. Can you take a look?

**Comment by larsoner** [comment]
... I'll just push a fix to #10348

**Pull request merged**